docs/tutorials/tut-design.md

```diff
@@ -0,0 +1,369 @@
+
+Customers typically use a combination of both: they use commands to automate processes through the playbooks and, when they are conducting investigations manually they run commands from the CLI to analyze the data.
+
+It's very important to understand how arguments and outputs work and what are the design best practices.
+
+#### Command Design
+
+Commands should be as atomic as possible (i.e. each command should ideally run a single API call to your product): this will simplify the handling of conditions where some calls fail and other succeeds: whenever possible such logic should be implemented in Playbooks rather than integrations.
+
+Make sure that commands run quickly and are non-blocking: a command should never take more than 2-3 seconds to run and return the information, or it would have significant performance impacts in XSOAR. **Avoid sleep() at all costs in your code**.
+
+If you have commands that need to run for longer periods of time, we recommend two approaches:
+- Make the commands asynchronous and implement a [Generic Polling](..//playbooks/generic-polling) mechanism. For example, if you need to run a search across your Endpoints, instead of having a single command that waits until the search is completed, you should implement three separate commands:
+  - A command that triggers the search and returns immediately a job id as output (i.e. `!helloworld-start-scan`).
+  - A command that checks the status of the job taking the job id as input (i.e. `!helloworld-scan-status`).
+  - A command that retrieves the results of a job when it's complete, taking the job id as input (i.e. `helloworld-scan-results`).
+  
+  A good example to refer to is the [HelloWorld](https://github.com/demisto/content/blob/master/Packs/HelloWorld/Integrations/HelloWorld/HelloWorld.py) integration.
+
+- Use [Long Running](../integrations/long-running) containers: this is suitable for services that need to keep a connection open for long time or need to open a listening TCP port. For example [Slack](https://github.com/demisto/content/tree/master/Packs/Slack/Integrations/Slack).
+
+There are cases where you want to build commands that perform generic well-known actions that are common across several use cases, for example Reputation commands (i.e. commands that return enrichment and reputation information about indicators, such as IPs). For those scenarios, we have standardized ways to define command names, inputs and outputs that will make interoperability much easier. For more information, check out the docs about [Generic Commands](../integrations/generic-commands-about) and [DBotScore](../integrations/dbot).
+
+#### Command Arguments
+
+Arguments are the inputs of your integration commands. They can be mandatory or not, and can have default and predefined values. More details [here](../integrations/yaml-file#commands).

```

Comment:
For the arguments we should suggest a good practice of making the inputs similar to what user required to provide in the original product. So for example in the product to create an incident user should choose a severity from 3 options High, Medium and Low, but the api supports 1,2,3 (numbers). Then the integration should support High, Medium and Low and convert these values in the integration. The user experience of integration should be familiar to the user